react/src/pages/admin/UserDetails.tsx

```diff
@@ -0,0 +1,204 @@
+                                            })
+                                        }
+                                    />
+                                }
+                            />
+                        </Grid>
+                        <Grid item xs={12}>
+                            <SecretDisplay
+                                title="MinIO Access Key"
+                                // TODO: Replace this with the actual secret
+                                secret={createHash("md5").update(props.user.username).digest("hex")}
+                            />
+                        </Grid>
+                        <Grid item xs={12}>
+                            <SecretDisplay
+                                title="MinIO Secret Key"
+                                // TODO: Replace this with the actual secret
+                                secret={createHash("md5").update(props.user.email).digest("hex")}
+                            />
+                        </Grid>
+                        <Grid item xs={12}>
+                            <NewPasswordForm
+                                passwords={{
+                                    password: newState.password!,
+                                    confirmPassword: newState.confirmPassword!,
+                                }}
+                                dispatch={dispatch}
+                            />
+                        </Grid>
+                    </Grid>
+                    <Grid item md={12} lg={6}>
+                        <Typography>
+                            <b>Group Management</b>
+                        </Typography>
+                        <ChipTransferList chips={tempGroups} setChips={setTempGroups} />
+                    </Grid>
+                </Grid>
+                <Toolbar className={classes.toolbar}>
+                    <Button
+                        color="secondary"
+                        variant="contained"
+                        className={classes.button}
+                        disabled // TODO: Re-enable when resetting MinIO Creds is ready
+                    >
+                        Reset MinIO Credentials
+                    </Button>
+                    <Button
+                        color="secondary"
+                        variant="contained"
+                        startIcon={<Delete />}
+                        className={classes.button}
+                        onClick={() => setConfirmDelete(true)}
+                    >
+                        Delete
+                    </Button>
+                    <div className={classes.grow} />
+                    <ButtonGroup>
+                        <Button
+                            variant="contained"
+                            onClick={() => {
+                                dispatch({ type: "set", payload: oldState });
+                            }}
+                        >
+                            Cancel
```

Comment:
I think all the changes should be discarded when the user clicks the cancel button